app/lib/common/widgets/spaces/select_space_form_field.dart

```diff
@@ -94,35 +90,18 @@ class SelectSpaceFormField extends ConsumerWidget {
   }
 
   Widget spaceBuilder(BuildContext context, WidgetRef ref, Widget? child) {
-    final spaceLoader = ref.watch(selectedSpaceDetailsProvider);
+    final space = ref.watch(selectedSpaceDetailsProvider);
     final currentId = ref.watch(selectedSpaceIdProvider);
-    return spaceLoader.when(
-      data: (space) {
-        if (space == null) return Text(currentId!);
-        return SpaceChip(
-          spaceId: space.roomId,
-          onTapOpenSpaceDetail: false,
-          useCompatView: useCompatView,
-          onTapSelectSpace: () {
-            if (useCompatView) selectSpace(context, ref);
-          },
-        );
-      },
-      error: (e, s) {
-        _log.severe('Failed to load the details of selected space', e, s);
-        return Text(L10n.of(context).loadingFailed(e));
-      },
-      loading: () => Skeletonizer(
-        child: Chip(
-          avatar: ActerAvatar(
-            options: AvatarOptions(
-              AvatarInfo(uniqueId: L10n.of(context).loading),
-              size: 24,
-            ),
+    return space.map(
```

Comment:
Can you link the relevant ticket for this error? Also I don't think we would need to introduce dependency for that. Like this can be handled by providing extension on generic type:
```
extension Let<T> on T? {
  R? let<R>(R? Function(T) op) {
    final T? value = this;
    return value == null ? null : op(value);
  }
}
```
and in UI:
```
child: space.let((s) => Column(
  mainAxisAlignment: MainAxisAlignment.start,
  crossAxisAlignment: CrossAxisAlignment.start,
  children: [
    Text(L10n.of(context).parentSpace),
    SpaceChip(
      spaceId: s.roomId,
      onTapOpenSpaceDetail: false,
    ),
  ],
)) ?? const SizedBox.shrink(),
```

Reply:
Bang operator is known as one of reasons that red screen appears in flutter app.
I just wanted to fix red screen issue.

Reply:
Your `Let` extension looks good.
BTW the shortage of `extension_nullable` is that it doesn't support async callback.
Your extension can break through it?

Reply:
I just posted this as en example. And yes, this may get confused with ```.map()``` fns.  About async thing, not sure it can handle that part but you can try adding that case ;)

Reply:
Fixed by https://github.com/acterglobal/a3/pull/2179/commits/a80e89005edc85169d05132c6974faf66f8e55e7